ziti/contexts.go

```diff
@@ -87,7 +87,12 @@ func NewContextWithOpts(cfg *Config, options *Options) (Context, error) {
 	}
 
 	if cfg.ID.Cert != "" && cfg.ID.Key != "" {
+		baseCreds := &edge_apis.BaseCredentials{}
```

Comment:
This if-block should not change. It is for backward compatibility for systems that rely only on the previous file configuration logic.

If an implementor wants to use the new credentials interface, I would expect them to create their own credentials object and include any headers on that field. I don't want to mix the usage of the `cfg.ID` field and the `cfg.Credentials`.

Reply:
I modified this conditional in an attempt to avoid side-effects. The problem is that BaseCredentials gets overwritten by the `NewIdentityCredentialsFromConfig` call.

Reply:
I've reverted this to the original conditional. The issue this conditional presents is that it creates a side-effect on the Credentials. My update to the code was an attempt to allow for the existing workaround behavior without it causing strange behavior. 

The workaround is to ensure that the SDK docs make it explicit that the credentials are not stable until _after_ the NewContext call is made.

Reply:
An example of what I'm talking about is here: https://github.com/openziti-test-kitchen/zssh/blob/d6425563e497e5f93b2225816cc4a2a753655aab/zsshlib/ssh.go#L329 -- I needed to move the update of Credentials to be _after_ the call to `NewContext` in order to avoid `NewContextWithOpts` from overwriting `BaseCredentials.Headers`